html_css/v2/main-course/intermediate_css/frameworks-preprocessors.md

```diff
@@ -0,0 +1,91 @@
+- How do you interact with the framework to get the look you want?
+- What are Bootstrap, Foundation, Tailwind and Bulma?
+- What other frameworks are find out there?
+- What are the disadvantages of using frameworks?
+
+- What is a preprocessor?
+- What are the advantages of using a preprocessor?
+- What does it mean that a preprocessor "compiles to CSS"?
+- What is SASS/SCSS?
+- How can you use SASS to nest CSS?
+- How can you use SASS to work with variables in your code?
+- What are the advantages of using mix-ins with SASS?
+- If you're not using Rails, how do you set up SASS?
+- What is LESS and how is it different from SASS?
+- What other preprocessors are out there?
+
+### Frameworks Overview
+
+Frameworks like [Bootstrap]](https://getbootstrap.com/) and [Foundation](https://get.foundation) do a lot of the heavy lifting of packaging up commonly used CSS code, even icons and interactions (like menu dropdowns), for you. They are designed to abstract away the process of coding intuitive, reusable, and responsive elements. They do this by creating these elements for you, and giving you the tools to inject styles into your HTML elements directly through class name keywords. All you have to do is understand how a framework expects you to lay out your site and which classes it uses to designate its particular batch of styles.
+
+While the two frameworks mentioned above are heavy hitters that have been around for a while, you should be aware that there are quite a few others at this point, as evidenced by all of the lists of popular frameworks on the web. Two other extremely popular frameworks are [Tailwind](https://tailwindcss.com/) and [Bulma](https://bulma.io/). There are many more out there.
+
+### Disadvantages of Frameworks
+
+Frameworks are great for rapidly producing sites with interfaces that end users can easily interact with. However, once you've taken a tour through some of the more popular ones, like Bootstrap, you'll start noticing an awful lot of similarities between a lot of sites you encounter, due to similar use of frameworks. Another concern with frameworks is that too many new developers jump into learning frameworks too early in their education; the prospect of not having to keep practicing writing vanilla CSS is very tempting. As a result, these developers do not get enough CSS practice under their belts to solidify the fundamentals of that very important language.
+
+Additionally, the process of overriding a framework's styling or debugging style issues on your page becomes very difficult if you haven't really mastered CSS fundamentals. It is imperative to understand what a framework is doing "under the hood" so you are equipped to handle these issues later (and trust us, you will).
+
+For the reasons above, **we recommend that you continue to focus on the fundamentals of CSS as you continue through the curriculum.** This lesson is merely designed to teach you _about_ frameworks and their role.
```

Comment:
this is a little heavy-hitting at this point lol... I'd go with a more neutral statement.

something like:

> Frameworks can help you get up and running quickly but can constrain you in the long run, and once you've started a project using one, it can be difficult to remove it.  Whether you use them for projects in the future is up to you!

Reply:
So I agree with you; this is lifted from an existing lesson and I actually tried to tone it down. Will replace with your suggestion.